src/server/cards/adventures/SwampHag.js

```diff
@@ -0,0 +1,27 @@
+import Card from 'cards/Card';
+
+export default class SwampHag extends Card {
+  static cost = new Card.Cost({ coin: 5 });
+  static types = new Set('Action', 'Attack', 'Duration');
+  async onPlay(player) {
+    this.ignoreCleanup = true;
+    player.addPersistentEffect('buy', this);
```

Comment:
I think players can block this attack, but only when this card is played. 
Basically the persistent effect can only trigger on players that didn't play a moat/lighthouse/champion